pay-api/src/pay_api/services/direct_pay_service.py

```diff
@@ -234,19 +281,92 @@ def _get_refund_token(cls):
+            elif refund_line.refund_type == RefundsPartialType.SERVICE_FEE.value:
+                DirectPayService._validate_refund_amount(refund_line.refund_amount, pli.service_fees)
+                service_fee_dist_id = fee_distribution.service_fee_distribution_code_id
+                service_fee_distribution = DistributionCodeModel.find_by_id(service_fee_dist_id)
+                revenue_account = DirectPayService._get_gl_coding(service_fee_distribution,
+                                                                  refund_line.refund_amount,
+                                                                  exclude_total=True)
+            refund_lines.append(RefundLineRequest(revenue_account, refund_line.refund_amount))
+            total += refund_line.refund_amount
+        return refund_lines, total
+
+    @classmethod
+    def _query_order_status(cls, invoice: InvoiceModel) -> OrderStatus:
+        """Request invoice order status from PAYBC."""
+        access_token: str = DirectPayService().get_token().json().get('access_token')
+        paybc_ref_number: str = current_app.config.get('PAYBC_DIRECT_PAY_REF_NUMBER')
+        paybc_svc_base_url = current_app.config.get('PAYBC_DIRECT_PAY_BASE_URL')
+        completed_reference = list(
+            filter(lambda reference: (reference.status_code == InvoiceReferenceStatus.COMPLETED.value),
+                   invoice.references))[0]
+        payment_url: str = \
+            f'{paybc_svc_base_url}/paybc/payment/{paybc_ref_number}/{completed_reference.invoice_number}'
+        payment_response = cls.get(payment_url, access_token, AuthHeaderType.BEARER, ContentType.JSON).json()
+        return Converter().structure(payment_response, OrderStatus)
+
+    @classmethod
+    def build_automated_refund_payload(cls, invoice: InvoiceModel, refund_partial: List[RefundPartialLine]):
         """Build payload to create a refund in PAYBC."""
         receipt = ReceiptModel.find_by_invoice_id_and_receipt_number(invoice_id=invoice.id)
         invoice_reference = InvoiceReferenceModel.find_by_invoice_id_and_status(
             invoice.id, InvoiceReferenceStatus.COMPLETED.value)
-        # Future: Partial refund support - This is backwards compatible
-        # refundRevenue: [{
-        #        'lineNumber': 1,
-        #        'refundAmount': 50.00,
-        #    }]
-        return {
+        refund_payload = {
             'orderNumber': int(receipt.receipt_number),
             'pbcRefNumber': current_app.config.get('PAYBC_DIRECT_PAY_REF_NUMBER'),
             'txnAmount': invoice.total,
             'txnNumber': invoice_reference.invoice_number
         }
+        if not refund_partial:
+            return refund_payload
+
+        refund_lines, total_refund = DirectPayService._build_refund_revenue_lines(refund_partial)
+        paybc_invoice = DirectPayService._query_order_status(invoice)
+        refund_payload.update({
```

Comment:
I probably could have used a `dataclass` for this, but I think i'll leave it for now